cmd/pmm-managed-starlark/main_test.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/percona/pmm-managed/services/checks"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRunChecks(t *testing.T) {
+	l := logrus.WithField("component", "pmm-managed-starlark")
+
+	testCases := []struct {
+		data   checks.StarlarkScriptData
+		err    bool
+		panics bool
+	}{
+		{
+			data: checks.StarlarkScriptData{
+				CheckName:         "invalid version",
+				Script:            "def check(): return []",
+				CheckVersion:      5,
+				QueryActionResult: "some result",
+			},
+			err:    true,
+			panics: false,
+		},
+		{
+			data: checks.StarlarkScriptData{
+				CheckName:         "invalid starlark syntax",
+				Script:            "@ + @",
+				CheckVersion:      1,
+				QueryActionResult: "some result",
+			},
+			err:    true,
+			panics: false,
+		},
+		{
+			data: checks.StarlarkScriptData{
+				CheckName:         "invalid query action result",
+				Script:            "def check(): return []",
+				CheckVersion:      1,
+				QueryActionResult: "some result",
+			},
+			err:    true,
+			panics: false,
+		},
+		{
+			data: checks.StarlarkScriptData{
+				CheckName:         "bad starlark script",
+				Script:            "def check(rows): return [1] * (1 << 30-1)",
+				CheckVersion:      1,
+				QueryActionResult: "\n\rVariable_name\n\x05Value\x12\x1c\n\t2\aversion\n\x0f2\r5.7.30-33-log\x12I\n\x112\x0fversion_comment\n422Percona Server (GPL), Release 33, Revision 6517692\x12%\n\x192\x17version_compile_machine\n\b2\x06x86_64\x12\x1f\n\x142\x12version_compile_os\n\a2\x05Linux\x12\x1a\n\x102\x0eversion_suffix\n\x062\x04-log",
```

Comment:
C'mon! 😆 We can do better there

Reply:
I just logged the result of the script that I was using locally and put it there, what changes do you want me to make here?

Reply:
To make it human-readable and -editable, of course. What prevents us from accepting data, encoding it, and passing to pmm-managed-starlark script where it will be decoded?

Reply:
so you want the string encoded separately and then sent with the json to the script where the json will be decoded and then the string can be further decoded before passing it to the starlark env?

I am not sure which encoding scheme to use to dealing with a string with ascii and non-printable characters mixed like this one. Do you have any suggestions for that?

Reply:
First of all, I want to be able to change test data. The current way of doing that does not allow it.

That binary data is a result of some function call, so why we not use the parameter of that function call as a test data, and the call that function ourselves in test?

Reply:
> That binary data is a result of some function call, so why we not use the parameter of that function call as a test data, and the call that function ourselves in test?

so the function call that gets me that data is `models.FindActionResultByID(s.db.Querier, resultID)` inside the `waitForResult` method, if I use `models.FindActionResultByID` arguments as the test data that means moving it to `pmm-managed-starlark` and supplying the arguments, the problem with that approach is that since `FindActionResultByID` gets called in a infinite for loop I will essentially have to rewrite the `waitForResults` method in the binary and for that I will need to send more data through STDIN like the context, pointer to `checks.Service` to be able to call `s.db.Delete(res)` afterwards, `resultCheckInterval` to run the ticker.

I just don't see how I can separate just the call to `models.FindActionResultByID` from the rest of the logic and move it to the binary 

One more thing after this review https://github.com/percona/pmm-managed/pull/451#discussion_r484532663  I moved the call to `agentpb.UnmarshalActionQueryResult` to the binary, but it broke the checks during QA, all of them failed while unmarshalling with `unexepected EOF` and when I moved the call to `agentpb.UnmarshalActionQueryResult` back to `waitForResults` method, the checks started working again.